02-microarray/pathway-analysis_microarray_07_gsea.Rmd

```diff
@@ -0,0 +1,443 @@
+These results are from a zebrafish microarray experiment we used for [differential expression analysis for two groups](https://alexslemonade.github.io/refinebio-examples/02-microarray/differential-expression_microarray_02_2-groups.html) using [`limma`](https://bioconductor.org/packages/release/bioc/html/limma.html) [@Ritchie2015].
+The table contains Ensembl gene IDs, log fold-changes for each group, and adjusted p-values (FDR in this case).
+We can identify differentially regulated genes by filtering these results and use this list as input to GSEA.
+
+Instead of using the URL below, you can use a file path to a TSV file with your desired gene list results.
+First we will assign the URL to its own variable called, `dge_url`. 
+
+```{r}
+# Define the url to your differential expression results file
+dge_url <- "https://refinebio-examples.s3.us-east-2.amazonaws.com/02-microarray/results/GSE71270/GSE71270_limma_results.tsv"
+```
+
+Read in the file that has differential expression results. 
+Here we are using the URL we set up above, but this can be a local file path instead _i.e._ you can replace `dge_url` in the code below with a path to file you have on your computer like: `file.path("results", "GSE71270_limma_results.tsv")`.
+
+```{r}
+# Read in the contents of your differential expression results file
+# `dge_url` can be replaced with a file path to a TSV file with your
+# desired gene list results
+dge_df <- readr::read_tsv(dge_url)
+```
+
+`read_tsv()` can read TSV files online and doesn't necessarily require you download the file first. 
+Let's take a look at what these contrast results from the differential expression analysis look like. 
+
+```{r}
+dge_df
+```
+
+## Getting familiar with `clusterProfiler`'s options
+
+Let's take a look at what organisms the package supports.
+
+```{r}
+msigdbr_species()
+```
+**DRAFT**
+MSigDB contains [8 different gene set collections](https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp) [@Subramanian2005].
+
+    H: hallmark gene sets
+    C1: positional gene sets
+    C2: curated gene sets
+    C3: motif gene sets
+    C4: computational gene sets
+    C5: GO gene sets
+    C6: oncogenic signatures
+    C7: immunologic signatures
+
+MSigDB includes a collection called Hallmark gene sets ([Liberzon *et al.* 2015](https://dx.doi.org/10.1016%2Fj.cels.2015.12.004))
+Here's an excerpt of the [collection description](https://www.gsea-msigdb.org/gsea/msigdb/collection_details.jsp#H):
+
+Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression.
+These gene sets were generated by a computational methodology based on identifying gene set overlaps and retaining genes that display coordinate expression.
+The hallmarks reduce noise and redundancy and provide a better delineated biological space for GSEA.
+We'll use the Hallmark collection for GSEA.
+Notably, there are only 50 gene sets included in this collection.
```

Comment:
To clarify a point from your initial comment:

> I used the hallmark genes dataset (rather than all genes)

> The genes of interest list step (step 6 of the original plan) did not seem beneficial/best suited for this analysis once I got into developing as the hallmark genes dataset is used, already limiting the genes shown in this example

By using the hallmark gene sets, you're limiting the number of _gene sets_ you are testing. These gene sets will contain differing numbers of genes. So Hallmark Gene Set A  could contain 35 genes, and Hallmark Gene Set B could have 135 genes. The preranked statistics that you pass to `GSEA()` are genome-wide (or at least everything that you have in the DGE table). The union of the genes that are in the Hallmark sets are fewer than you'll have in your DGE table, but the same could be said about KEGG pathways or one of the other collections.

Because each gene set included is a test, by reducing the number of tests you run, you lower your multiple hypothesis testing burden (and your run time, too!). I would not conceptualize this as using fewer genes overall, though.

Reply:
Gotcha!